python/foglamp/services/core/api/backup_restore.py

```diff
@@ -7,113 +7,148 @@
+    """ Returns the details of a backup
 
-    :Example: curl -X GET  http://localhost:8082/foglamp/backup/1
+    :Example: curl -X GET http://localhost:8081/foglamp/backup/1
     """
     backup_id = request.match_info.get('backup_id', None)
     if not backup_id:
         raise web.HTTPBadRequest(reason='Backup id is required')
-    else:
-        try:
-            backup_id = int(backup_id)
-        except ValueError:
-            raise web.HTTPBadRequest(reason='Invalid backup id')
     try:
-        # TODO : Fix after actual implementation
-        Backup.get_backup_details.return_value = \
-            {"date": '2017-08-30 04:05:10.382', "status": "running"}
-    except Backup.DoesNotExist:
+        backup_id = int(backup_id)
+        backup = Backup(connect.get_storage())
+        backup_json = backup.get_backup_details(backup_id)
+
+        resp = {"status": _get_status(int(backup_json["status"])),
+                'id': backup_json["id"],
+                'date': backup_json["ts"]
+                }
+
+    except ValueError:
+        raise web.HTTPBadRequest(reason='Invalid backup id')
+    except exceptions.DoesNotExist:
         raise web.HTTPNotFound(reason='Backup with {} does not exist'.format(backup_id))
+    except Exception as ex:
+        raise web.HTTPException(reason=(str(ex)))
+
+    return web.json_response(resp)
 
-    _resp = Backup.get_backup_details(id=backup_id)
-    _resp["id"] = backup_id
-    return web.json_response(_resp)
 
 async def delete_backup(request):
-    """
-    Delete a backup
+    """ Delete a backup
 
-    :Example: curl -X DELETE  http://localhost:8082/foglamp/backup/1
+    :Example: curl -X DELETE http://localhost:8081/foglamp/backup/1
     """
     backup_id = request.match_info.get('backup_id', None)
     if not backup_id:
         raise web.HTTPBadRequest(reason='Backup id is required')
-    else:
-        try:
-            backup_id = int(backup_id)
-        except ValueError:
-            raise web.HTTPBadRequest(reason='Invalid backup id')
-        try:
-            # TODO : Fix after actual implementation
-            Backup.delete_backup.return_value = "Backup deleted successfully"
-        except Backup.DoesNotExist:
-            raise web.HTTPNotFound(reason='Backup with {} does not exist'.format(backup_id))
+    try:
+        backup_id = int(backup_id)
+        backup = Backup(connect.get_storage())
+        backup.delete_backup(backup_id)
+        return web.json_response({'message': "Backup deleted successfully"})
+    except ValueError:
+        raise web.HTTPBadRequest(reason='Invalid backup id')
+    except exceptions.DoesNotExist:
+        raise web.HTTPNotFound(reason='Backup with {} does not exist'.format(backup_id))
+    except Exception as ex:
+        raise web.HTTPException(reason=str(ex))
+
 
-        _resp = Backup.delete_backup(id=backup_id)
-        return web.json_response({'message': _resp})
+def _get_status(status_code):
+    if status_code not in range(1, 6):
```

Comment:
In your case below case fails:
```
if 6 not in range(1, 6):
    print ("YES")
```    
YES

So, range check should (1, 7 ) as per Status enum